### DISCUSSION TITLE
Optional support for columns with dots "dot.s"

### DISCUSSION THREAD

**Discussion by daimor** [sqlalchemy/sqlalchemy#8993] [Ideas]
Could you make using dots in names optional in test, with requirements?
In our database there is no way to escape dot, it's forbidden to use

https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/testing/suite/test_dialect.py#L382

```
Delimited identifier expected, delimited identifier containing invalid character '.' found ^   "dot.s"
```

**Comment by zzzeek** [comment]
There's no restriction on characters in bind parameter names because we are supposed to escape them fully without ever exposing them to the database.  in this case, dots are not currently in the list of characters that we escape by default.  note a char like `%` is never passed:

```py
from sqlalchemy import create_engine, bindparam, select, event

e = create_engine("postgresql://scott:tiger@localhost/test", echo='debug')

@event.listens_for(e, "before_cursor_execute")
def before_cursor_execute(conn, cus, stmt, param, *arg, **kw):
    print(f"actual statement {stmt}")
    print(f"actual params {param}")

stmt = select(1).where(bindparam("per%cent") == 1)

with e.connect() as conn:
    print(conn.scalar(stmt, {"per%cent": 1}))
```

output:

```
actual statement SELECT 1 
WHERE %(perPcent)s = %(param_1)s
actual params {'perPcent': 1, 'param_1': 1}

```

for your dialect, at the moment you can emulate oracle's support here:  https://github.com/sqlalchemy/sqlalchemy/blob/5bb48511a126b66ed06abf76d706ab707afafbf1/lib/sqlalchemy/dialects/oracle/cx_oracle.py#L697    (though much simpler as you likely dont have the odd quoting thing that Oracle does)

where you use an alternate _BIND_TRANSLATE_RE and _BIND_TRANSLATE_CHARS

I do think we should make these two regexes pluggable by dialects.    I can support PRs or efforts towards making these two variables into per-dialect settings.

**Reply by zzzeek** [reply]
I will add pluggable support hopefully today, see #8994

**Reply by daimor** [reply]
I don't see that Oracle's case somehow would help 

```
[%msg: < Delimited identifier expected, delimited identifier containing invalid character '.' found ^   "dot.s">]
[SQL: 
CREATE TABLE t (
        id IDENTITY NOT NULL, 
        "dot.s" VARCHAR(50) NOT NULL, 
        PRIMARY KEY (id)
)
```
So, it fails just during create table, and I did not find a good way to get around it

```
test_round_trip_same_named_column[dot.s] FAILED
test_standalone_bindparam_escape[dot.s] PASSED
test_standalone_bindparam_escape_expanding[dot.s] PASSED
```

**Reply by zzzeek** [reply]
oh it's creating the table also.   OK will think of something

**Reply by zzzeek** [reply]
it's just the one test.  You can simply skip this test for now:

```py

from sqlalchemy.testing.suite import *

from sqlalchemy.testing.suite import DifficultParametersTest as _DifficultParametersTest

class DifficultParametersTest(_DifficultParametersTest):

    def test_round_trip_same_named_column(self):
        pass

```

**Reply by daimor** [reply]
Sure, already did it this way
```python
class DifficultParametersTest(_DifficultParametersTest):
    @_DifficultParametersTest.tough_parameters
    def test_round_trip_same_named_column(
        self, paramname, connection, metadata
    ):
        if '.' in paramname:
            pass
        else:
            return super().test_round_trip_same_named_column(paramname, connection, metadata)

```

**Reply by zzzeek** [reply]
that's fine

this specific test is mostly redundant, the actual escaping of parameter names is also tested by the other two tests in that suite without using an oddly-named column.  so it's safe to just skip it for the thing that is being tested here.
